mem.go

```diff
@@ -583,28 +583,16 @@ func (e *Encoder) Encode(m *Message) error {
 		e.hdrbuf = e.hdrbuf[:hdrSize]
 	}
 	marshalStreamHeader(e.hdrbuf, sizes)
-	if err := e.write(e.hdrbuf); err != nil {
-		return err
-	}
+	bufs := make([][]byte, 1+nsegs)
```

Comment:
GH is not letting me add a comment farther up... while you're visiting this code, it seems like traversing the segments twice (using `segmentSizes`) isn't really necessary.  Mind rolling that all into one?